packages/forma-36-react-components/src/components/TextField/TextField.tsx

```diff
@@ -47,118 +48,104 @@ const defaultProps: Partial<TextFieldProps> = {
   width: 'full',
 };
 
-export class TextField extends Component<TextFieldProps, TextFieldState> {
-  static defaultProps = defaultProps;
-
-  state = { value: this.props.value || '', initialValue: this.props.value };
+export const TextField = (props: TextFieldProps) => {
+  const {
+    validationMessage,
+    className,
+    textInputProps,
+    testId,
+    width,
+    formLabelProps,
+    textLinkProps,
+    labelText,
+    helpText,
+    textarea,
+    countCharacters,
+    required,
+    onChange,
+    onBlur,
+    onFocus,
+    value,
+    name,
+    id,
+    ...otherProps
+  } = props;
+  const [valueState, setValueState] = useState<string | undefined>(value);
 
   // Store a copy of the value in state.
   // This is used by this component when the `countCharacters`
   // option is on
-  handleOnChange = (evt: ChangeEvent) => {
-    this.setState({ value: (evt.target as HTMLInputElement).value });
-    if (this.props.onChange) this.props.onChange(evt);
+  const handleOnChange = (
```

Comment:
`useCallback` would be nice here, I imagine performance can drop quite a bit with quick typing otherwise?